drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1270,6 +1261,17 @@ static int uarte_instance_init(struct device *dev,
 
 	nrf_uarte_disable(uarte);
 
+#ifdef CONFIG_UART_ASYNC_API
+	const struct uarte_nrfx_config *cfg = get_dev_config(dev);
+
+	data->async->rx_total_byte_cnt = 0;
+	data->async->rx_total_user_byte_cnt = 0;
+	if (hw_rx_counting_enabled(data)) {
+		nrfx_timer_clear(&cfg->timer);
+	} else {
+		data->async->rx_cnt.cnt = 0;
+	}
```

Comment:
This should be moved to `uarte_nrfx_rx_counting_init()`. It fits better there.
Besides, `nrfx_timer_clear()` should be called after `nrfx_timer_init()`, and only when initialization of the timer was successful.

Reply:
done, noticed also that timer is clear there after successful `nrfx_timer_init`.

Reply:
since time is cleared and counters in the structure are initialized to 0 from the start, i have removed resetting at all.